pilot/pkg/serviceregistry/kube/controller/discoverycontrollers.go

```diff
@@ -64,17 +74,25 @@ func (c *Controller) initDiscoveryNamespaceHandlers(
 			newNs := new.(*v1.Namespace)
 			membershipChanged, namespaceAdded := discoveryNamespacesFilter.NamespaceUpdated(oldNs.ObjectMeta, newNs.ObjectMeta)
 			if membershipChanged {
-				var handleFunc func() error
-				if namespaceAdded {
-					handleFunc = func() error {
+				handleFunc := func() error {
+					var event model.Event
+					if namespaceAdded {
+						event = model.EventAdd
 						c.handleSelectedNamespace(endpointMode, newNs.Name)
-						return nil
-					}
-				} else {
-					handleFunc = func() error {
+					} else {
+						event = model.EventDelete
 						c.handleDeselectedNamespace(kubeClient, endpointMode, newNs.Name)
-						return nil
 					}
```

Comment:
This should be abstracted in to a function and call it from both places.

May be should we call this from handleSelectedNamespace and handleDeselectedNamespace so that we do not miss calling this?

Reply:
>May be should we call this from handleSelectedNamespace and handleDeselectedNamespace so that we do not miss calling this?

This will introduce redundant push triggered when multi namespaces need to be handled in case  discoverySelectors changed

Reply:
But, they will be merged any way because of debouncing. But you can leave it but better to abstract it in to a func and call it.

Reply:
Would think about abstraction

Reply:
moved `namespaceDiscoveryHandlers`